src/js/components/AppListItemComponent.jsx

```diff
@@ -19,6 +21,80 @@ var AppListItemComponent = React.createClass({
     model: React.PropTypes.object.isRequired
   },
 
+  getInitialState: function () {
+    return {numberOfVisibleLabels: -1};
+  },
+
+  componentDidMount: function () {
+    // Avoid referencing window from Node context
+    if (global.window != null) {
+      window.addEventListener("resize", this.handleResize);
+      window.addEventListener("focus", this.handleResize);
+    }
+    this.updateNumberOfVisibleLabels();
+
+  },
+
+  componentDidUpdate: function (prevProps) {
+    if (!_.isEqual(this.props, prevProps)) {
+      this.updateNumberOfVisibleLabels();
+    }
+  },
+
+  componentWillUnmount: function () {
+    if (global.window != null) {
+      window.removeEventListener("resize", this.handleResize);
+      window.removeEventListener("focus", this.handleResize);
+    }
+  },
+
+  shouldComponentUpdate: function (nextProps, nextState) {
+    var props = this.props;
+
+    if (nextState.numberOfVisibleLabels === -1 && !props.model.isGroup) {
+      this.updateNumberOfVisibleLabels();
+      return false;
+    }
+
+    return !_.isEqual(props, nextProps) ||
```

Comment:
OK, this needs to go out. At a small scale this may be effective, but considering 60.000 tasks in the props.model, let this become a bottleneck.
You could just remove this or move the labels logic into an component and just comparing the model.labels.